test/io/test_io.py

```diff
@@ -368,6 +368,7 @@ def test_jwt_secret_external_file_reload(tmp_path, defaultenv):
         "PGRST_DB_CHANNEL_ENABLED": "true",
         "PGRST_DB_CONFIG": "false",
         "PGRST_DB_ANON_ROLE": "postgrest_test_anonymous",  # required for NOTIFY
+        "PGRST_JWT_CACHE_MAX_LIFETIME": "86400",
```

Comment:
How about a dedicated test for this case?

It's not clear from looking at the code that this addition proves the fix, plus looks like this new test will be needed for https://github.com/PostgREST/postgrest/pull/4008 too.